src/Network/Transport/TCP.hs

```diff
@@ -929,7 +929,12 @@ handleConnectionRequest transport socketClosed (sock, sockAddr) = handle handleE
               -- Clear the socket of the unsupported handshake data.
               _ <- recvExact sock handshakeLength
               handleVersioned
-    handleVersioned
+    -- The handshake must complete within the optional timeout duration.
+    let connTimeout = transportConnectTimeout (transportParams transport)
+    outcome <- maybe (fmap Just) System.Timeout.timeout connTimeout handleVersioned
```

Comment:
Doesn't this make useless checking the timeout inside `handleConnectionRequestV0`?